src/azul/objects.py

```diff
@@ -64,3 +64,17 @@ def __new__(_cls, *args, **kwargs):
             return _self
 
         cls.__new__ = __new__
+
+
+# Vendored from boltons==19.3.0
+class classproperty(object):
```

Comment:
Does PyCharm infer the return type correctly with this?

Does [this](https://github.com/python/mypy/issues/2563#issuecomment-523837469) technique have any advantages?

Reply:
> Does PyCharm infer the return type correctly with this?

No, not for my test:

```
class C:
    @classproperty
    def foo(cls) -> int:
        return 42
    

c = "a" + C.foo
d = "a" + 42
```

> Does this technique have any advantages?

PyCharm does not correctly infer types with this form either. Since this form requires two decorators, I think it is inferior. Also the two decorators raise a different, "incompatible decorator" warning in pycharm.

[This implementation](https://github.com/python/mypy/issues/2563#issue-195305499) is functionally the same as the boltons one, but with better type annotations. Even though the type annotations don't allow inference with PyCharm, I'll _steal_ this one instead.

Reply:
The main reason we switch to functools.cached_property is to enable PyCharm type inference.

I think we should not encourage the use of `@classproperty` since there is no implementation that we know of that allows PyCharm type inference of the return type. For that reason, I think we should completely remove the decorator and eliminate the one use in `health.py`.